proxy/src/main/java/net/md_5/bungee/UserConnection.java

```diff
@@ -140,6 +140,7 @@
     @Getter
     @Setter
     private ForgeServerHandler forgeServerHandler;
+    private boolean disconnecting;
```

Comment:
This should be part of ChannelWrapper. You also haven't rationalized its existence.